homeassistant/components/otbr/config_flow.py

```diff
@@ -85,10 +89,12 @@ async def _connect_and_set_dataset(self, otbr_url: str) -> None:
                 _LOGGER.debug(
                     "not importing TLV with channel %s", thread_dataset_channel
                 )
+                pan_id = generate_random_pan_id()
                 await api.create_active_dataset(
                     python_otbr_api.ActiveDataSet(
                         channel=allowed_channel if allowed_channel else DEFAULT_CHANNEL,
-                        network_name="home-assistant",
+                        network_name=compose_default_network_name(pan_id),
+                        pan_id=pan_id,
```

Comment:
What does setting the `pan_id` do?

Reply:
The OTBR will use that value as the PAN ID when forming the network. So far, a random PAN ID gets generated by the OTBR code (specifically by https://github.com/openthread/openthread/blob/a0b9ac74f1b467df5722ea8613a2dcb690b93fd4/src/core/mac/mac_types.cpp#L52C23-L52C38). Because we want the PAN ID part of the name, we have to generate it on our end and pass it while forming the network, to make sure the OTBR takes this PAN ID instead of (another) randomly generated one.